buildengine/engine.go

```diff
@@ -92,13 +100,15 @@ func (e *Engine) startSchemaSync(ctx context.Context) func(ctx context.Context,
 		switch msg.ChangeType {
 		case ftlv1.DeploymentChangeType_DEPLOYMENT_ADDED, ftlv1.DeploymentChangeType_DEPLOYMENT_CHANGED:
 			sch, err := schema.ModuleFromProto(msg.Schema)
-			if err == nil {
+			if err != nil {
 				return err
 			}
 			e.controllerSchema.Store(sch.Name, sch)
+			e.schemaChanges.Publish(schemaChange{ChangeType: msg.ChangeType, Module: sch})
```

Comment:
I think you'll also want to Publish the schema to the topic when you build locally, for the situation where you're building offline? Or I guess we don't have a "dev" mode where we only build, so maybe not.

Reply:
Yeah I think that's true. I think the dev mode is assuming hot reloading with real controller, etc.